src/Imdb/Title.php

```diff
@@ -477,29 +477,46 @@ public function comment_split() {
  #-------------------------------------------------------[ Recommendations ]---
   /**
    * Get recommended movies (People who liked this...also liked)
-   * @return array recommendations (array[imdbid,title,year])
+   * @return array recommendations (array[title,imdbid,year,endyear,type])
    * @see IMDB page / (TitlePage)
    */
-  public function movie_recommendations() {
-    if (empty($this->movierecommendations)) {
-      $doc = new \DOMDocument();
-      @$doc->loadHTML($this->getPage("Title"));
-      $xp = new \DOMXPath($doc);
-      $cells = $xp->query("//div[@id=\"title_recs\"]/div[@class=\"rec_overviews\"]/div[@class=\"rec_overview\"]/div[@class=\"rec_details\"]");
-      foreach ($cells as $cell) {
-        preg_match('!tt(\d+)!',$cell->getElementsByTagName('a')->item(0)->getAttribute('href'),$ref);
-        $movie['title'] = trim($cell->getElementsByTagName('a')->item(0)->nodeValue);
-        $movie['imdbid'] = $ref[1];
-        preg_match('!(\d+)!',$cell->getElementsByTagName('span')->item(0)->nodeValue,$ref);
-        if (!isset($ref[1])) {
-          preg_match('!(\d+)!',$cell->getElementsByTagName('span')->item(1)->nodeValue,$ref);
-        }
-        $movie['year'] = $ref[1];
-        $this->movierecommendations[] = $movie;
-      }
-    }
-    return $this->movierecommendations;
-  }
+public function movie_recommendations() {
+	if (empty($this->movierecommendations)) {
+		$doc = new \DOMDocument();
+		@$doc->loadHTML($this->getPage("Title"));
+		$xp = new \DOMXPath($doc);
+		if ($cells = $xp->query("//div[@id=\"title_recs\"]/div[@class=\"rec_overviews\"]/div[@class=\"rec_overview\"]/div[@class=\"rec_details\"]/div[@class=\"rec-info\"]/div[@class=\"rec-jaw-upper\"]/div[@class=\"rec-title\"]")){
+			foreach ($cells as $cell) {
+				if(preg_match('!tt(\d+)!',$cell->getElementsByTagName('a')->item(0)->getAttribute('href'),$ref)){
+					$movie['title'] = trim($cell->getElementsByTagName('a')->item(0)->nodeValue);
+					$movie['imdbid'] = $ref[1];
+					if($span = $cell->getElementsByTagName('span')->item(0)->nodeValue){
+						$years = preg_replace('/[^0-9]/','',$span);
+						$type = trim(preg_replace('/[^a-z\s]/i','',strip_tags($span)));
+						if(mb_strlen(trim($years)) >4){
+							$movie['year'] = trim(substr($years, 0, 4));
+							$movie['endyear'] = trim(substr($years, 4));
+							$movie['type'] = $type;
```

Comment:
I don't like this field. I can't figure out what the reasoning behind it is, and it's not filled far more often than it is

Reply:
For example on https://www.imdb.com/title/tt0303461/ you might expect most of the recommendations to be 'tv series' or something like that .. but battlestar galactica (the series) does not say tv series, band of brothers does not say miniseries however sherlock does say tv series.
It's far too inconsistent to provide any value.

Reply:
I think you are talking about the type field?

It simply contains any text that would be between the brackets. So i didn't make it inconsistent, it is exactly like it is on imdb.
If you think it is inconsistent, oke i understand but don't blame me haha
The value of the type field can be up for debate, i can leave it out if you will

Reply:
Sorry, this is my first time using this code review functionality on github. I selected the last line that is showing in these code snippets.

I'm suggesting that the field is unlikely to provide any value to anyone because there isn't any information you can derive from it. You can't know that it's a film, not a film, or that it's a TV series.

Reply:
Oke thanks for the review.
Jep i included it, but i agree with your argurments that it is inconsistent and therefor not reliable info.


I will remove it

Reply:
Reworked, the type field is removed